pkg/volume/azure_dd/attacher.go

```diff
@@ -0,0 +1,291 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure_dd
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/cloudprovider"
+	"k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/util/keymutex"
+	"k8s.io/kubernetes/pkg/util/mount"
+	"k8s.io/kubernetes/pkg/volume"
+)
+
+type azureDiskAttacher struct {
+	host    volume.VolumeHost
+	manager azureManager
+}
+
+var _ volume.Attacher = &azureDiskAttacher{}
+
+var _ volume.AttachableVolumePlugin = &azureDataDiskPlugin{}
+
+const (
+	checkSleepDuration = time.Second
+)
+
+// serialize concurrent attach
+var attachMutex = keymutex.NewKeyMutex()
+
+func (plugin *azureDataDiskPlugin) NewAttacher() (volume.Attacher, error) {
+	azure, err := getAzureDiskManager(plugin.host.GetCloudProvider())
+	if err != nil {
+		glog.Infof("failed to get azure provider")
+		return nil, err
+	}
+
+	return &azureDiskAttacher{
+		host:    plugin.host,
+		manager: azure,
+	}, nil
+}
+
+func (attacher *azureDiskAttacher) Attach(spec *volume.Spec, hostName string) (string, error) {
+	volumeSource, err := getVolumeSource(spec)
+	if err != nil {
+		glog.Infof("failed to get azure disk spec")
```

Comment:
delete (and all other similar glog.Infof on errors)  returning an error should be sufficient, logging additionally causes spam.